mrp_workorder_sequence/models/mrp_workorder.py

```diff
@@ -1,10 +1,24 @@
 # Copyright 2019-20 ForgeFlow S.L. (https://www.forgeflow.com)
 # License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class MrpWorkOrder(models.Model):
     _inherit = "mrp.workorder"
 
     sequence = fields.Integer()
+
+    @api.model_create_multi
+    def create(self, values_list):
+        for values in values_list:
+            if values.get("sequence"):
+                continue
+            last_production_wo = self.search(
+                [("production_id", "=", values.get("production_id"))],
+                order="sequence DESC",
+                limit=1,
+            )
+            if last_production_wo:
+                values["sequence"] = last_production_wo.sequence + 1
```

Comment:
NITPICKING

It seems this code will give you more than 1 record with the exact same sequence in case you create multiple workorders for the same `production_id` at once.

Not sure this is acceptable or not (out of context).

Also, performance wise it's not a good idea to have a `search` within a loop.

Maybe you could identify all the `vals` that need a new sequence before-hand, extract their `production_id` and search for all of them at once using `read_group` and the `sequence:max` aggregator.

This also allows you to set unique sequences in the case described above, as you'd have everything in memory by the time you enter the loop, you can maintain a dict mapping with `{production_id: last_sequence}`, and increment that number in-memory as you process them.

```py
    for values in vals_list:
        if values.get("sequence"):
            continue
        production_id = values["production_id"]
        last_sequence_by_production.setdefault(production_id, 0)
        last_sequence_by_production[production_id] += 1
        values["sequence"] = last_sequence_by_production[production_id]
```

Reply:
Still working on this one

Reply:
Solved in fixup commit below.